### ISSUE TITLE
Edit Connection doesn't work correctly

### ISSUE THREAD

**Issue by mtaylorfsmb** [microsoft/azuredatastudio#24011]
Type: <b>Bug</b>

I have a set of connections saved in some groups. Because of the encryption changes in the connection string I need to adjust the connection properties. The `Browse` button in the `Connection` window that appears when you try to connect from a query window gives you the option to edit the connection.

When you edit the connection it takes you to the `Recent` section with the connection details loaded. This section also gives you an edit connection option. In either case you can make changes to the connection but attempting to connect then fails because the connection has already been saved.

1. Select a connection that has already been saved from the `Browse` section of the `Connect` UI when connecting from an editor window.
2. Click the pencil to edit the connection. Alternatively do the same thing from the `Recent` list.
3. In the connection details change something.
4. There is no "save" option so click `Connect`.
5. ADS shows an error that the connection already exists.

![image](https://github.com/microsoft/azuredatastudio/assets/38144840/f964e34a-e051-4422-aece-d53c9458b468)

There doesn't seem to be a way to actually save changes to an existing connection, or at least the UI would indicate as such.

Azure Data Studio version: azuredatastudio 1.45.0 (8eff468100f02157f2219335f7a566d14451ca49, 2023-07-24T18:22:12.106Z)
OS version: Windows_NT x64 10.0.19044
Restricted Mode: No
Preview Features: Disabled
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz (8 x 3600)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.92GB (17.57GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
poor-sql-formatter|WSR|0.1.0

</details>

**Comment by cheenamalhotra** [comment]
As the message suggests, you seem to have an existing connection with exact same properties (including advanced options) - have you verified your connection profiles?

**Comment by mtaylorfsmb** [comment]
@cheenamalhotra Yes there is an existing connection. That's the point. If you go to a saved connection and click the Edit pencil next to it then you're modifying an existing connection. There is no "save" option so if you make any changes then you have to click Connect to "save". At that point you get an error irrelevant of whether you made any change or not. I have noticed that it does save the changes, it appears, but then pops up the error message that the connection already exists. 

The easiest equivalent example is opening an existing file in notepad and clicking save. If notepad displays a "file already exists" error then it doesn't make sense because you're editing an existing file so it should already exist.

**Comment by cheenamalhotra** [comment]
I mean if you have a connection in Connections pane > Server tree view, clicking on Connect that modifies a different connection to match properties with an existing connection will try to add separate connection in the servers list, which won't be allowed to prevent duplicate connections.

If you delete any existing duplicate connections from server tree view, you should be able to edit saved connections normally.
From my testing I don't notice any abnormal behavior where editing connection falsely throws error without an actual conflicting connection being present.

Can you provide specific steps to follow to reproduce what you're seeing in a clean session?

**Comment by mtaylorfsmb** [comment]
@cheenamalhotra Yes you are correct. The saved connections had 3 different copies of the same connection. 2 of them were identical and one differed by just the database being connected to. I deleted the dups and the problem went away.

I'm unclear how the dups got in there though as I did a combination of connecting new (because the recent list got wiped with this update as well) and trying to edit the existing connections. Somehow ADS let the dups get created and then failed on editing them. Nevertheless that seems unrelated to this specific issue.
